### PR TITLE
Fix RemoteDisconnect issue: Add sleep / async call / backoff

### LINKED ISSUE #96: RemoteDisconnected issue when storing prediction results from two submissions at once
```
STDERR: 2021-01-28T06:17:27.983046800Z 	    nlpsandbox/cli:edge \
STDERR: 2021-01-28T06:17:27.983050991Z 	    community \
STDERR: 2021-01-28T06:17:27.983055336Z 	    store-annotations \
STDERR: 2021-01-28T06:17:27.983059531Z 	    --annotation_json \
STDERR: 2021-01-28T06:17:27.983063776Z 	    /var/lib/cwl/stg7777c434-8ded-4e9b-bb58-991cc060f40b/predictions.json \
STDERR: 2021-01-28T06:17:27.983068171Z 	    --annotation_store_id \
STDERR: 2021-01-28T06:17:27.983072344Z 	    submission-9710064 \
STDERR: 2021-01-28T06:17:27.983076312Z 	    --data_node_host \
STDERR: 2021-01-28T06:17:27.983080555Z 	    http://10.23.54.142/api/v1/ \
STDERR: 2021-01-28T06:17:27.983084741Z 	    --dataset_id \
STDERR: 2021-01-28T06:17:27.983088899Z 	    2014-i2b2-20201203
STDERR: 2021-01-28T06:17:27.983093033Z 	Traceback (most recent call last):
STDERR: 2021-01-28T06:17:27.983097218Z 	  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
STDERR: 2021-01-28T06:17:27.983101764Z 	    httplib_response = self._make_request(
STDERR: 2021-01-28T06:17:27.983105977Z 	  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 445, in _make_request
STDERR: 2021-01-28T06:17:27.983110544Z 	    six.raise_from(e, None)
STDERR: 2021-01-28T06:17:27.983114727Z 	  File "<string>", line 3, in raise_from
STDERR: 2021-01-28T06:17:27.983119646Z 	  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 440, in _make_request
STDERR: 2021-01-28T06:17:27.983124259Z 	    httplib_response = conn.getresponse()
STDERR: 2021-01-28T06:17:27.983128494Z 	  File "/usr/local/lib/python3.8/http/client.py", line 1347, in getresponse
STDERR: 2021-01-28T06:17:27.983132915Z 	    response.begin()
STDERR: 2021-01-28T06:17:27.983137041Z 	  File "/usr/local/lib/python3.8/http/client.py", line 307, in begin
STDERR: 2021-01-28T06:17:27.983141351Z 	    version, status, reason = self._read_status()
STDERR: 2021-01-28T06:17:27.983149191Z 	  File "/usr/local/lib/python3.8/http/client.py", line 276, in _read_status
STDERR: 2021-01-28T06:17:27.983153638Z 	    raise RemoteDisconnected("Remote end closed connection without"
STDERR: 2021-01-28T06:17:27.983157954Z 	http.client.RemoteDisconnected: Remote end closed connection without response
```

### PR DESCRIPTION
**Pull request by thomasyu888**
* Use asynchronous call to store annotations
* Added retry logic for urllib3

### CODE REVIEW

**Comment by tschaffter** [comment]
@thomasyu888 I would like to understand better the cause of the issue so we can find a reliable solution.

**Comment by thomasyu888** [comment]
> @thomasyu888 I would like to understand better the cause of the issue so we can find a reliable solution.

@tschaffter My guess of why this worked is that I am waiting a second before doing another rest call.  All of the code that actually returns the response in the server is abstracted away and is part of `connexion` (unmodifiable).  My experiment of adding a sleep in the controller itself caused the same `RemoteDisconnect` issue, which led me to think that the client could be making too many calls too fast.

Ive been trying to search more about this issue, but I can't seem to find more than what I've already found.

**Comment by tschaffter** [comment]
@thomasyu888 A better approach would be to block/loop over the response until the status code 201 is received. This way we always wait enough time instead of assuming that the request will be completed within 1 second (response time depends on hardware and connection bandwidth).

The ideal solution would be to send the [requests asynchronously](https://stackoverflow.com/a/9189249). In JavaScript/TypeScript, I would use [mergeMap](https://www.learnrxjs.io/learn-rxjs/operators/transformation/mergemap) to send a bunch of requests at once, which is one instruction that would then return once all the requests have completed. The uWSGI server of the data node is [configured to process one request at a time](https://github.com/nlpsandbox/data-node/blob/develop/server/uwsgi.ini#L9) but this could be changed in a later revision to decrease processing time (e.g. pulling clinical notes concurrently, sending requests to tools concurrently, etc.).

Could you give a try to the first solution proposed (wait for 201 response)?

**Comment by thomasyu888** [comment]
@tschaffter Thanks for the resources!  I took a look at where I would even look for a response == 201, the issue is the `api_instance.create_annotation(...` function returns the json object already, so I would have to look deeper in the code.  Upon investigate, they seemed to have generated `request` functions within the classes which is great but also, there are more response codes than just 201.

I'm trying something different.  Apparently I can add a `async_req=True` command to use asynchronous HTTP calls.  lets see if that works.

File: nlpsandboxclient/client.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 """NLP data node client that interacts with the SDK datanodeclient"""
+import time
```
Comment (thomasyu888) [inline nlpsandboxclient/client.py:2]:
```suggestion
```

**Review by tschaffter — APPROVED** [review]
Looks great!

**Pull request merged**